batch/batch/batch.py

```diff
@@ -541,9 +542,10 @@ def is_successful(self):
         return self._state == 'Success'
 
     async def mark_unscheduled(self):
-        await self._delete_pod()
-        if not self._cancelled or self.always_run:
+        if self._status == 'Running':
             await self.set_state('Ready')
+
+        if self._state == 'Ready' and (not self._cancelled or self.always_run):
             await self._create_pod()
```

Comment:
If you try this without deleting the pod, it will fail, no?

Reply:
Ah, I see.  I thought mark_unscheduled meant "there is no pod," but it really means "schedule or reschedule".  Will fix.